pkg/oauthserver/oauth/registry/registry_test.go

```diff
@@ -1,4 +1,4 @@
-package registry
+package registry_test // prevent import cycle
```

Comment:
We should remove the cycle. The `pkg/oauthserver/osinserver/registrystorage` pkg shouldn't use an interface defined in the consumer but should define it's own minimal one.

Reply:
The most clear would be removing the interface entirely; it has no purpose and obscures what's going on. Otherwise, keep the interface definition in `registrystorage`.

Reply:
The interface is used for mocking during tests.  Will move it back.

Reply:
There's nothing to mock. The test can be updated to use the real type rather than a mock.

Reply:
OK - dropped the interface and rewrote test to work without it.